api/v1alpha6/conversion.go

```diff
@@ -94,6 +94,20 @@ func restorev1alpha6MachineSpec(previous *OpenStackMachineSpec, dst *OpenStackMa
 	dst.Ports = previous.Ports
 }
 
+func restorev1alpha6ClusterStatus(previous *OpenStackClusterStatus, dst *OpenStackClusterStatus) {
+	// PortOpts.SecurityGroups have been removed in v1alpha7
+	// We restore the whole PortOpts/Networks since they are anyway immutable.
+	if previous.ExternalNetwork != nil {
```

Comment:
Couple of interesting notes:

Firstly, it looks like ExternalNetwork might be mutable! I doubt this is intentional, though.

Secondly, the only reason you have to do this at all is because the Network type is used in too many places and doesn't make sense in most/all of them. Cleaning it up may be the largest remaining v1beta1 task.

Given the above I suspect it may be ok to merge this anyway. I'll try to get back to it properly later: I haven't had time to follow these thoughts through fully yet.

Reply:
No, it's not mutable: the webhook prevents changes. The second point still stands, but not relevant to this PR.